### ISSUE TITLE
usage of extent report 2.40 load config

### ISSUE THREAD

**Issue by Madhavsri** [anshooarora/extentreports#187]
Please let me know how can I use load config option in extent report 2.40 version

After giving the path for config file it is not able to fetch the config file 

com.relevantcodes.extentreports.Configuration <init>
INFO: Configurating report from file:/E:/Parallel-%20TestNG%20programitically/Parallel_FrameWork/lib/extentreports-java-2.40.0.jar!/com/relevantcodes/extentreports/resources/extent-config.xml

Hi Anshoo thanks for your prompt reply but my config.xml file is kept at a different path 
which I have mentioned in the loadconfig command below
//reporter.loadConfig(new File("E:\\Parallel- TestNG programitically\\Parallel_FrameWork\\extent-config.xml"));
Please suggest.

**Comment by anshooarora** [comment]
As per the message, the config file was consumed.

**Comment by vertissachin** [comment]
Hi,
Ansho i am getting the error in console as follows

PM com.relevantcodes.extentreports.Configuration <init>
INFO: Configurating report from file:/C:/Users/Sameer/.m2/repository/com/relevantcodes/extentreports/2.40/extentreports-2.40.jar!/com/relevantcodes/extentreports/resources/extent-config.xml

and also the report.html is not getting created not ablt to find that why it is giving this error

**Comment by anshooarora** [comment]
Its not an error, its just INFO telling you the inbuild config file is being used.

The report is not generated due to some other error, I will have to see your code and stack-trace.  If this issue is unrelated to @Madhavsri, please open a new ticket.

**Comment by anshooarora** [comment]
@Madhavsri - are you still experiencing this issue?  Note: the message you are getting is a INFO telling you the standard config file was used.

**Comment by vertissachin** [comment]
Hi thanks for your reply my above issue is got resolve but when my test script fails the screen shot is not getting attached to the extent report .

How to use the extent report for creating result of multiple test case . also can we get the testing logs into our extent report like testing report gives us log in case of failure

**Comment by vertissachin** [comment]
Hi ,

When my script is getting failed for element due to time out then screen shot is not  captured some time and the report is not getting generated

also the screen shot is not getting attached to report/not showing clearly 

Code used for capturing screen in teardown() method 

            String screenshot = TestUtilities.captureScreenShot(webDriver,result.getName());
            String img = test.addScreenCapture(screenshot);
            test.log(LogStatus.FAIL,"Verify Discovery screen",img);
![img](https://cloud.githubusercontent.com/assets/15407829/11707861/2ff6e804-9f2f-11e5-9861-d30bc5e4e709.jpg)

i am using the extent report 2.40 . 

also looks like the extent report 2.05 is better that the 2.40

**Comment by vertissachin** [comment]
extent report 2.05 UI generated by default config 
![img1](https://cloud.githubusercontent.com/assets/15407829/11707912/c36bfc96-9f2f-11e5-962c-63faaa09d78d.png)

**Comment by Madhavsri** [comment]
Hi Anshoo,

I am running parallel implementation of extent report.
As in the parallel implementation we have factory class and each test creating reporter instance from Factory class.
Reporter is initialized in the following manner in the factory class
String path=".\\Report"+"\\"+TestBaseClass.LocalreportDirName+"\\"+"Report.html";
			if (!ReadObject.getFailTCFlag()){
			
			reporter = new ExtentReports(path, true, DisplayOrder.NEWEST_FIRST);
			}else {
				path=".\\Report"+"\\"+ xls.getLastRunResultFolderName()+"\\Rerun"+"\\"+"Rerun.html";
				reporter = new ExtentReports(path, true, DisplayOrder.NEWEST_FIRST);
			} 
but reporter object is not getting initialized and no report is also getting generated.
the behavior is same even if I do not supply configuration via loadonfig

**Comment by anshooarora** [comment]
@Madhavsri  - There should be only 1 instance of Extent for your session, unless you want to create multiple reports.

You can view examples here to manage the reporting and test instances: <a href='https://github.com/anshooarora/extentreports/tree/master/java/extentreports/src/test/java/com/relevantcodes/extentreports/common'>linky</a>.

**Comment by harishmajithiya** [comment]
Hi Anshoo,
I am facing the same issue as @Madhavsri. It works without Maven, but with maven it gives error...

See below the details:
Exception in thread "main" java.lang.NoClassDefFoundError: org/w3c/dom/ElementTraversal
	at java.lang.ClassLoader.defineClass1(Native Method)
	at java.lang.ClassLoader.defineClass(Unknown Source)
	at java.security.SecureClassLoader.defineClass(Unknown Source)
	at java.net.URLClassLoader.defineClass(Unknown Source)
	at java.net.URLClassLoader.access$100(Unknown Source)
	at java.net.URLClassLoader$1.run(Unknown Source)
	at java.net.URLClassLoader$1.run(Unknown Source)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.net.URLClassLoader.findClass(Unknown Source)
	at java.lang.ClassLoader.loadClass(Unknown Source)
	at sun.misc.Launcher$AppClassLoader.loadClass(Unknown Source)
	at java.lang.ClassLoader.loadClass(Unknown Source)
	at org.apache.xerces.parsers.AbstractDOMParser.startDocument(Unknown Source)
	at org.apache.xerces.impl.dtd.XMLDTDValidator.startDocument(Unknown Source)
	at org.apache.xerces.impl.XMLDocumentScannerImpl.startEntity(Unknown Source)
	at org.apache.xerces.impl.XMLVersionDetector.startDocumentParsing(Unknown Source)
	at org.apache.xerces.parsers.XML11Configuration.parse(Unknown Source)
	at org.apache.xerces.parsers.XML11Configuration.parse(Unknown Source)
	at org.apache.xerces.parsers.XMLParser.parse(Unknown Source)
	at org.apache.xerces.parsers.DOMParser.parse(Unknown Source)
	at org.apache.xerces.jaxp.DocumentBuilderImpl.parse(Unknown Source)
	at javax.xml.parsers.DocumentBuilder.parse(Unknown Source)
	at com.relevantcodes.extentreports.Configuration.getConfigurationMap(Configuration.java:38)
	at com.relevantcodes.extentreports.Report.loadConfig(Report.java:218)
	at com.relevantcodes.extentreports.ExtentReports.loadConfig(ExtentReports.java:271)
	at com.relevantcodes.extentreports.ExtentReports.<init>(ExtentReports.java:80)
	at com.relevantcodes.extentreports.ExtentReports.<init>(ExtentReports.java:213)
	at com.vauto.reporting.TestReport.beginReport(TestReport.java:88)
	at com.vauto.executor.Executor.main(Executor.java:24)
Caused by: java.lang.ClassNotFoundException: org.w3c.dom.ElementTraversal
	at java.net.URLClassLoader.findClass(Unknown Source)
	at java.lang.ClassLoader.loadClass(Unknown Source)
	at sun.misc.Launcher$AppClassLoader.loadClass(Unknown Source)
	at java.lang.ClassLoader.loadClass(Unknown Source)
	... 29 more


Thanks,
Harish

**Comment by anshooarora** [comment]
@harishmajithiya  -  can you share any code that can help me reproduce this?  and you are getting this error via the maven dependency?

**Comment by harishmajithiya** [comment]
Looks like this is isolated from maven, even without maven I am facing issue, it started after I used 2.40.. 

for code side of thing, i am doing nothing but initializing. (only once)

ExtentReports extent = new ExtentReports(CurrentRun.extentReportLocation + "HTMLReport.html",true);

I was ok to give me own extent-config.xml in the next line of code, but it's not even going there..

Thanks,
Harish

**Comment by anshooarora** [comment]
You are missing this class `java.lang.NoClassDefFoundError: org/w3c/dom/ElementTraversal`

What version of Java?  Can you do a clean build of your project and try again?

**Comment by harishmajithiya** [comment]
Thanks Anshoo for looking into this and prompt response. 

I am using 1.8, and clean build did not help too.  I needed to add following dependency to make this work, Not sure what it has to do with.. but it's working now.


	<dependency>
			<groupId>xml-apis</groupId>
			<artifactId>xml-apis</artifactId>
			<version>1.4.01</version>
		</dependency>

**Comment by anshooarora** [comment]
I have not had any reports of this issue, or having to add the dependency that you've added.  See <a href='https://github.com/anshooarora/extentreports/blob/master/java/extentreports/src/main/java/com/relevantcodes/extentreports/Configuration.java#L18'>here</a> - its just native Java that is required to traverse nodes.  In my dev or test projects, this dependency is not associated either.

Glad you got it to work though, still unsure what causes this.

**Comment by shaynl** [comment]
I had the same issue with ver 2.40 and adding this dependency solved it. Thanks!

**Comment by amaralrfl** [comment]
Hi anshooarora,
I'm having the same problem to start this issue. I'm using load config like this:

public static ExtentReports getReportInstance(String filePath) {
		if (extent == null) {
			extent = new ExtentReports(filePath, false);			
			extent.loadConfig(new File("my-source-path/.../reportextent-config.xml"));
		}

But the aplication get the path:
my-maven-repository/.../com/relevantcodes/extentreports/2.40/extentreports-2.40.jar!/com/relevantcodes/extentreports/resources/extent-config.xml


The application is not picking up the folder I specified . Would you help me? Thanks for you help!!

**Comment by anshooarora** [comment]
Its not able to find the file with the path you specified.  Can you check your path and try again?

```java
// Should it be (notice the dots):
extent.loadConfig(new File("my-source-path/../reportextent-config.xml"));

// instead of
extent.loadConfig(new File("my-source-path/.../reportextent-config.xml"));
```

**Comment by amaralrfl** [comment]
I check and was not really specifying the right path. Thank you for your help!

**Comment by vertissachin** [comment]
Hello I am getting issue while generating HTML file and file is getting
created empty , I think you have resolved this issue but when it will
available on latest jar or when latest jar version will be released?

**Comment by karthikbhatnagar** [comment]
You should share complete/ wrapper code to be analysed

**Comment by vertissachin** [comment]
Hi Ansho,

i have a question how to generate the reports when we run the test in
multibrowser at the same time
actulay we are running the test on multiple browser using xml file like

 <test name="WIN8_Chrome46">
    <parameter name="browser" value="chrome" />
    <parameter name="platform" value="WIN8" />
    <parameter name="version" value="46" />
    <classes>
        <class name="test1" />
    </classes>
</test>

<test name="WIN8_Firefox37">
    <parameter name="browser" value="firefox" />
    <parameter name="platform" value="WIN8" />
    <parameter name="version" value="37" />
    <classes>
        <class name="test2" />
    </classes>
</test>

but after running the test i am getting the only last browser result in
html file.
so could you please help to generate report for multibrowser run

thanks ... !!!

On Mon, Jan 11, 2016 at 12:00 PM, Karthik <notifications@github.com> wrote:

> You should share complete/ wrapper code to be analysed
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/anshooarora/extentreports/issues/187#issuecomment-170443811>
> .
>



-- 

Thanks and Regards,
Sachin Pathare
Software Engineer QA

*Vertis Microsystems LLP*
*E*:  sachin.pathare@vertisinfotech.com
*A*:  B-407 - Fourth Floor, Teerth Technospace IT Park, Off
Mumbai-Bangalore Highway, Baner, Pune, India 411045
*O*: +91 (0) 20 6720 7602 | M:+91 9011454470
*W*: www.vertisinfotech.com
LinkedIn: http://www.linkedin.com/company/vertis-microsystems-llp

*Disclaimer*: The information contained in this e-mail and any attachments
is confidential, may be privileged and is intended solely for the use of
the named addressee. Access to this e-mail by any other person is not
authorized. If you are not the intended recipient, you should not disclose,
copy, distribute, take any action or rely on it and you should, please,
notify the sender by reply.
<https://drive.google.com/a/vertisinfotech.com/folderview?id=0B1ywHyjXHOhOeHRjcllaeGdIU2s&usp=gmail>

**Comment by anshooarora** [comment]
Are you utilizing the same Extent instance for both?  Ending both tests at end?  Would like to see your code to know for sure what's going on.

**Comment by vertissachin** [comment]
Hi,

Here is the my code where i had used for report generation and



public class AdvReports {

    public static ExtentReports reporter;
    public static Map<Long, String> threadToExtentTestMap = new
HashMap<Long, String>();
    public static Map<String, ExtentTest> nameToTestMap = new
HashMap<String, ExtentTest>();

    public static String browserName;

    public  static synchronized ExtentReports getExtentReport(String browser) {

        if (reporter == null) {
            // you can get the file name and other parameters here from a
            // config file or global variable
            browserName = browser;
            reporter = new
ExtentReports("./reports/TestSuitRunReport_"+browser+".html", true,
DisplayOrder.NEWEST_FIRST);
            System.out.println("Test report is generating for browser
:"+ browserName);
        }
        return reporter;
    }

    public  static synchronized ExtentTest getTest(String testName,
String testDescription) {


        // if this test has already been created return
        if (!nameToTestMap.containsKey(testName)) {
            Long threadID = Thread.currentThread().getId();
            ExtentTest test =
getExtentReport(browserName).startTest(testName, testDescription);
            nameToTestMap.put(testName, test);
            threadToExtentTestMap.put(threadID, testName);
        }
        return nameToTestMap.get(testName);
    }

    public  static synchronized ExtentTest getTest(String testName) {

        return getTest(testName, "");
    }
    public  static synchronized ExtentTest getTest() {

        Long threadID = Thread.currentThread().getId();

        if (threadToExtentTestMap.containsKey(threadID)) {
            String testName = threadToExtentTestMap.get(threadID);
            return nameToTestMap.get(testName);
        }

        return null;
    }

    public  static synchronized void closeTest(String testName) {


        if (!testName.isEmpty()) {
            ExtentTest test = getTest(testName);
            getExtentReport(browserName).endTest(test);
        }
        //browserName =null;
    }

    public  static  synchronized void closeTest(ExtentTest test) {

        if (test != null) {
            getExtentReport(browserName).endTest(test);
        }
        //browserName =null;
    }

    public  static synchronized void closeTest()  {

        ExtentTest test = getTest();
        closeTest(test);
    }

    public  static void closeReport() {
        if (reporter != null) {
            System.out.println("Report is generated for browser: " +
browserName);
            reporter.flush();
            reporter.close();
        }
    }
    public  static synchronized void AddBrowserInfo(String name,String
version) {

        if (name != null) {
            reporter.addSystemInfo("Browser Name", name);
            reporter.addSystemInfo("Browser Version",version);
        }
    }


}







Below is the class where i had instantiated reports.


public class VastMidrollAutoplayFalseInitialTime10Tests {

    private WebDriver webDriver;
    private static String TestName;
    private static String ipaddress;
    private static String v4Version;
    private static ExtentTest test;
    private static ExtentReports report;

    DesiredCapabilities capabilities = new DesiredCapabilities();

    @Parameters({ "browser", "version", "platform"})
    @BeforeTest
    public void beforeTest(@Optional String browser, @Optional String
version, @Optional String platform) throws Exception {

        DesiredCapabilities dr = null;
        TestName=this.getClass().getSimpleName();
        report = AdvReports.getExtentReport(browser);
        AdvReports.AddBrowserInfo(browser, version);
        ipaddress = Utils.readPropertyOrEnv("ipaddress", "10.11.69.126:5555");
        ipaddress = "http://"+ipaddress+"/wd/hub";
        v4Version=Utils.readPropertyOrEnv("v4Version","candidate/latest");
        v4Version= URLEncoder.encode(v4Version, "UTF-8");

        if (browser.equalsIgnoreCase("saucelab")) {
            webDriver = Utils.setUpSauceLabConnection(webDriver);
        }else {
            dr=Utils.browserSetup(browser);
            if (version != null) dr.setVersion(version);
            dr.setCapability(CapabilityType.PLATFORM, platform);
            webDriver = new RemoteWebDriver(new URL(ipaddress), dr);//
Start Selenium Grid
        }
    }

    @DataProvider(name = "testUrls")
    public static Object[][] getTestData() {
        return Utils.parseJsonInput_DataProvider(TestName,"Deeplook_Vast_AMF.json",v4Version);
    }

    @Test(dataProvider = "testUrls")
    public void verifyMidrollAutoplayFalseInitialTime10(String
Description, String testURL) throws Exception {


      }
    @AfterMethod
    public void tearDown(ITestResult result) throws Exception {

        if(result.getStatus()==ITestResult.FAILURE)
        {
            String screenshot =
TestUtilities.captureScreenShot(webDriver,result.getName());
            String img = test.addScreenCapture(screenshot);
            test.log(LogStatus.FAIL,"Verify Midroll Autoplay false
Initial time test",img);
            test.log(LogStatus.INFO,result.getThrowable());
        }
        System.out.println("Tests are done. Quitting webdriver");
        System.out.println("Test URL " + webDriver.getCurrentUrl());
        report.endTest(test);
        AdvReports.closeTest();
        webDriver.close();
        webDriver.quit();
    }
    @AfterSuite
    public void afterSuite() {
        AdvReports.closeReport();
    }
}


On Tue, Jan 19, 2016 at 8:32 PM, Anshoo Arora <notifications@github.com>
wrote:

> Are you utilizing the same Extent instance for both? Ending both tests at
> end? Would like to see your code to know for sure what's going on.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/anshooarora/extentreports/issues/187#issuecomment-172879132>
> .
>



-- 

Thanks and Regards,
Sachin Pathare
Software Engineer QA

*Vertis Microsystems LLP*
*E*:  sachin.pathare@vertisinfotech.com
*A*:  B-407 - Fourth Floor, Teerth Technospace IT Park, Off
Mumbai-Bangalore Highway, Baner, Pune, India 411045
*O*: +91 (0) 20 6720 7602 | M:+91 9011454470
*W*: www.vertisinfotech.com
LinkedIn: http://www.linkedin.com/company/vertis-microsystems-llp

*Disclaimer*: The information contained in this e-mail and any attachments
is confidential, may be privileged and is intended solely for the use of
the named addressee. Access to this e-mail by any other person is not
authorized. If you are not the intended recipient, you should not disclose,
copy, distribute, take any action or rely on it and you should, please,
notify the sender by reply.
<https://drive.google.com/a/vertisinfotech.com/folderview?id=0B1ywHyjXHOhOeHRjcllaeGdIU2s&usp=gmail>

**Comment by vertissachin** [comment]
also if i add  following code in each file getting lots of error message in
console/cmd

@AfterSuite
public void afterSuite() {
    AdvReports.closeReport();
}



On Wed, Jan 20, 2016 at 10:17 AM, Sachin Pathare <
sachin.pathare@vertisinfotech.com> wrote:

> Hi,
>
> Here is the my code where i had used for report generation and
>
>
>
> public class AdvReports {
>
>     public static ExtentReports reporter;
>     public static Map<Long, String> threadToExtentTestMap = new HashMap<Long, String>();
>     public static Map<String, ExtentTest> nameToTestMap = new HashMap<String, ExtentTest>();
>
>     public static String browserName;
>
>     public  static synchronized ExtentReports getExtentReport(String browser) {
>
>         if (reporter == null) {
>             // you can get the file name and other parameters here from a
>             // config file or global variable
>             browserName = browser;
>             reporter = new ExtentReports("./reports/TestSuitRunReport_"+browser+".html", true, DisplayOrder.NEWEST_FIRST);
>             System.out.println("Test report is generating for browser :"+ browserName);
>         }
>         return reporter;
>     }
>
>     public  static synchronized ExtentTest getTest(String testName, String testDescription) {
>
>
>         // if this test has already been created return
>         if (!nameToTestMap.containsKey(testName)) {
>             Long threadID = Thread.currentThread().getId();
>             ExtentTest test = getExtentReport(browserName).startTest(testName, testDescription);
>             nameToTestMap.put(testName, test);
>             threadToExtentTestMap.put(threadID, testName);
>         }
>         return nameToTestMap.get(testName);
>     }
>
>     public  static synchronized ExtentTest getTest(String testName) {
>
>         return getTest(testName, "");
>     }
>     public  static synchronized ExtentTest getTest() {
>
>         Long threadID = Thread.currentThread().getId();
>
>         if (threadToExtentTestMap.containsKey(threadID)) {
>             String testName = threadToExtentTestMap.get(threadID);
>             return nameToTestMap.get(testName);
>         }
>
>         return null;
>     }
>
>     public  static synchronized void closeTest(String testName) {
>
>
>         if (!testName.isEmpty()) {
>             ExtentTest test = getTest(testName);
>             getExtentReport(browserName).endTest(test);
>         }
>         //browserName =null;
>     }
>
>     public  static  synchronized void closeTest(ExtentTest test) {
>
>         if (test != null) {
>             getExtentReport(browserName).endTest(test);
>         }
>         //browserName =null;
>     }
>
>     public  static synchronized void closeTest()  {
>
>         ExtentTest test = getTest();
>         closeTest(test);
>     }
>
>     public  static void closeReport() {
>         if (reporter != null) {
>             System.out.println("Report is generated for browser: " + browserName);
>             reporter.flush();
>             reporter.close();
>         }
>     }
>     public  static synchronized void AddBrowserInfo(String name,String version) {
>
>         if (name != null) {
>             reporter.addSystemInfo("Browser Name", name);
>             reporter.addSystemInfo("Browser Version",version);
>         }
>     }
>
>
> }
>
>
>
>
>
>
>
> Below is the class where i had instantiated reports.
>
>
> public class VastMidrollAutoplayFalseInitialTime10Tests {
>
>     private WebDriver webDriver;
>     private static String TestName;
>     private static String ipaddress;
>     private static String v4Version;
>     private static ExtentTest test;
>     private static ExtentReports report;
>
>     DesiredCapabilities capabilities = new DesiredCapabilities();
>
>     @Parameters({ "browser", "version", "platform"})
>     @BeforeTest
>     public void beforeTest(@Optional String browser, @Optional String version, @Optional String platform) throws Exception {
>
>         DesiredCapabilities dr = null;
>         TestName=this.getClass().getSimpleName();
>         report = AdvReports.getExtentReport(browser);
>         AdvReports.AddBrowserInfo(browser, version);
>         ipaddress = Utils.readPropertyOrEnv("ipaddress", "10.11.69.126:5555");
>         ipaddress = "http://"+ipaddress+"/wd/hub";
>         v4Version=Utils.readPropertyOrEnv("v4Version","candidate/latest");
>         v4Version= URLEncoder.encode(v4Version, "UTF-8");
>
>         if (browser.equalsIgnoreCase("saucelab")) {
>             webDriver = Utils.setUpSauceLabConnection(webDriver);
>         }else {
>             dr=Utils.browserSetup(browser);
>             if (version != null) dr.setVersion(version);
>             dr.setCapability(CapabilityType.PLATFORM, platform);
>             webDriver = new RemoteWebDriver(new URL(ipaddress), dr);// Start Selenium Grid
>         }
>     }
>
>     @DataProvider(name = "testUrls")
>     public static Object[][] getTestData() {
>         return Utils.parseJsonInput_DataProvider(TestName,"Deeplook_Vast_AMF.json",v4Version);
>     }
>
>     @Test(dataProvider = "testUrls")
>     public void verifyMidrollAutoplayFalseInitialTime10(String Description, String testURL) throws Exception {
>
>
>       }
>     @AfterMethod
>     public void tearDown(ITestResult result) throws Exception {
>
>         if(result.getStatus()==ITestResult.FAILURE)
>         {
>             String screenshot = TestUtilities.captureScreenShot(webDriver,result.getName());
>             String img = test.addScreenCapture(screenshot);
>             test.log(LogStatus.FAIL,"Verify Midroll Autoplay false Initial time test",img);
>             test.log(LogStatus.INFO,result.getThrowable());
>         }
>         System.out.println("Tests are done. Quitting webdriver");
>         System.out.println("Test URL " + webDriver.getCurrentUrl());
>         report.endTest(test);
>         AdvReports.closeTest();
>         webDriver.close();
>         webDriver.quit();
>     }
>     @AfterSuite
>     public void afterSuite() {
>         AdvReports.closeReport();
>     }
> }
>
>
> On Tue, Jan 19, 2016 at 8:32 PM, Anshoo Arora <notifications@github.com>
> wrote:
>
>> Are you utilizing the same Extent instance for both? Ending both tests at
>> end? Would like to see your code to know for sure what's going on.
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/anshooarora/extentreports/issues/187#issuecomment-172879132>
>> .
>>
>
>
>
> --
>
> Thanks and Regards,
> Sachin Pathare
> Software Engineer QA
>
> *Vertis Microsystems LLP*
> *E*:  sachin.pathare@vertisinfotech.com
> *A*:  B-407 - Fourth Floor, Teerth Technospace IT Park, Off
> Mumbai-Bangalore Highway, Baner, Pune, India 411045
> *O*: +91 (0) 20 6720 7602 | M:+91 9011454470
> *W*: www.vertisinfotech.com
> LinkedIn: http://www.linkedin.com/company/vertis-microsystems-llp
>
> *Disclaimer*: The information contained in this e-mail and any
> attachments is confidential, may be privileged and is intended solely for
> the use of the named addressee. Access to this e-mail by any other person
> is not authorized. If you are not the intended recipient, you should not
> disclose, copy, distribute, take any action or rely on it and you should,
> please, notify the sender by reply.
>
> <https://drive.google.com/a/vertisinfotech.com/folderview?id=0B1ywHyjXHOhOeHRjcllaeGdIU2s&usp=gmail>
>



-- 

Thanks and Regards,
Sachin Pathare
Software Engineer QA

*Vertis Microsystems LLP*
*E*:  sachin.pathare@vertisinfotech.com
*A*:  B-407 - Fourth Floor, Teerth Technospace IT Park, Off
Mumbai-Bangalore Highway, Baner, Pune, India 411045
*O*: +91 (0) 20 6720 7602 | M:+91 9011454470
*W*: www.vertisinfotech.com
LinkedIn: http://www.linkedin.com/company/vertis-microsystems-llp

*Disclaimer*: The information contained in this e-mail and any attachments
is confidential, may be privileged and is intended solely for the use of
the named addressee. Access to this e-mail by any other person is not
authorized. If you are not the intended recipient, you should not disclose,
copy, distribute, take any action or rely on it and you should, please,
notify the sender by reply.
<https://drive.google.com/a/vertisinfotech.com/folderview?id=0B1ywHyjXHOhOeHRjcllaeGdIU2s&usp=gmail>

**Comment by anshooarora** [comment]
I am not exactly sure what's happing in your `AdvReports` class.  It creates a new file based on the browser passed, and the 1st browser passed is the file that will be created.  A better design would be to add browser-name as category for test using:

```
test.assignCategory(browserName);
```

Also see the following utility classes:

* [ExtentManager](https://github.com/anshooarora/extentreports/blob/master/java/extentreports/src/test/java/com/relevantcodes/extentreports/common/ExtentManager.java)
* [ExtentTestManager](https://github.com/anshooarora/extentreports/blob/master/java/extentreports/src/test/java/com/relevantcodes/extentreports/common/ExtentTestManager.java)

You are also calling `closeReport()` in `afterMethod` which is why you get errors.  Once you close the report, no more events can be written.  See [here](http://extentreports.relevantcodes.com/java/version2/docs.html#close-extent).

**Comment by vertissachin** [comment]
Hi,

i tried using test.assignCategory(browserName); it is creating different
category for browser but logs comes only for one
*please check attached html file.*

i am using selenium grid to distribute the test cases on multiple browser
and multiple platforms.
what if i am running test cases on multiple browser on two different
platoform (*please check following xml code*)

i am running that file using maven command.

and following code i used.

*MyClassFile*

@BeforeClass
public void beforeTest(@Optional String browser, @Optional String
version, @Optional String platform) throws Exception {

    DesiredCapabilities dr = null;
    System.out.println("Starting
VastMidrollAutoplayFalseInitialTime10Tests in browser " + browser+ "
and Platform "+ platform );
    TestName=this.getClass().getSimpleName();
    report = AdvReports.getExtentReport();
    AdvReports.AddBrowserInfo(browser, version);

    *test = AdvReports.getTest(TestName,browser);*

    ipaddress = Utils.readPropertyOrEnv("ipaddress", "10.11.69.126:5555");
    ipaddress = "http://"+ipaddress+"/wd/hub";
    v4Version=Utils.readPropertyOrEnv("v4Version","candidate/latest");
    v4Version= URLEncoder.encode(v4Version, "UTF-8");

    if (browser.equalsIgnoreCase("saucelab")) {
        webDriver = Utils.setUpSauceLabConnection(webDriver);
    }else {
        dr=Utils.browserSetup(browser);
        if (version != null) dr.setVersion(version);
        dr.setCapability(CapabilityType.PLATFORM, platform);
        webDriver = new RemoteWebDriver(new URL(ipaddress), dr);//
Start Selenium Grid
    }
}




*ReportGenerationFile*



package com.ooyala.util; import java.util.HashMap; import java.util.Map;
import com.relevantcodes.extentreports.DisplayOrder; import
com.relevantcodes.extentreports.ExtentReports; import
com.relevantcodes.extentreports.ExtentTest; import
javax.xml.bind.SchemaOutputResolver; import java.io.File; public class
AdvReports { public static ExtentReports reporter; public static Map<Long,
String> threadToExtentTestMap = new HashMap<Long, String>(); public static
Map<String, ExtentTest> nameToTestMap = new HashMap<String, ExtentTest>();
public static String browserName; public static ExtentReports
getExtentReport() { if (reporter == null) { // you can get the file name
and other parameters here from a // config file or global variable reporter
= new ExtentReports("./reports/TestSuitRunReport.html", true,
DisplayOrder.NEWEST_FIRST); } return reporter; } public static ExtentTest
getTest(String testName, String testDescription) { // if this test has
already been created return if (!nameToTestMap.containsKey(testName)) {
Long threadID = Thread.currentThread().getId(); ExtentTest test =
getExtentReport().startTest(testName, testDescription);
System.out.println("Test report is generating for browser :"+
testDescription); * test.assignCategory(testDescription);*
nameToTestMap.put(testName, test); threadToExtentTestMap.put(threadID,
testName); } return nameToTestMap.get(testName); } public static ExtentTest
getTest(String testName) { return getTest(testName, ""); } public static
ExtentTest getTest() { Long threadID = Thread.currentThread().getId(); if
(threadToExtentTestMap.containsKey(threadID)) { String testName =
threadToExtentTestMap.get(threadID); return nameToTestMap.get(testName); }
return null; } public static void closeTest(String testName) { if
(!testName.isEmpty()) { ExtentTest test = getTest(testName);
getExtentReport().endTest(test); } } public static void
closeTest(ExtentTest test) { if (test != null) {
getExtentReport().endTest(test); } } public static void closeTest() {
ExtentTest test = getTest(); closeTest(test); } public static void
closeReport() { if (reporter != null) { reporter.flush(); reporter.close();
} } public static void AddBrowserInfo(String name,String version) { if
(name != null) { reporter.addSystemInfo("Browser Name", name);
reporter.addSystemInfo("Browser Version",version); } } }


testng XML File

<!DOCTYPE suite SYSTEM "http://testng.org/testng-1.0.dtd"> <suite
name="Selenium Alice Deep Multiple Browser Tests" parallel="tests"
thread-count="4"> <test name="WIN8_Chrome46"> <parameter name="browser"
value="chrome" /> <parameter name="platform" value="WIN8" /> <parameter
name="version" value="47" /> <classes> <class
name="VastMidrollAutoplayInitialTime10Tests" /> </classes> </test> <test
name="WIN8_Firefox37"> <parameter name="browser" value="firefox" />
<parameter name="platform" value="WIN8" /> <parameter name="version"
value="43.0" /> <classes> <class
name="VastMidrollAutoplayInitialTime10Tests" /> </classes> </test> <test
name="Windows_Firefox37"> <parameter name="browser" value="firefox" />
<parameter name="platform" value="Windows" /> <parameter name="version"
value="37" /> <classes> <class name="VastMidrollAutoplayInitialTime10Tests"
/> </classes> </test> <test name="WINDOWS_chrome46"> <parameter
name="browser" value="chrome" /> <parameter name="platform" value="WINDOWS"
/> <parameter name="version" value="46" /> <classes> <class
name="VastMidrollAutoplayInitialTime10Tests" /> </classes> </test> <test
name="MAC_Safari8"> <parameter name="browser" value="safari" /> <parameter
name="platform" value="MAC" /> <parameter name="platform" value="MAC" />
<parameter name="version" value="8.0" /> <classes> <class
name="VastMidrollAutoplayInitialTime10Tests" /> </classes> </test> </suite>




On Wed, Jan 20, 2016 at 9:39 PM, Anshoo Arora <notifications@github.com>
wrote:

> I am not exactly sure what's happing in your AdvReports class. It creates
> a new file based on the browser passed, and the 1st browser passed is the
> file that will be created. A better design would be to add browser-name as
> category for test using:
>
> test.assignCategory(browserName);
>
> Also see the following utility classes:
>
>    - ExtentManager
>    <https://github.com/anshooarora/extentreports/blob/master/java/extentreports/src/test/java/com/relevantcodes/extentreports/common/ExtentManager.java>
>    - ExtentTestManager
>    <https://github.com/anshooarora/extentreports/blob/master/java/extentreports/src/test/java/com/relevantcodes/extentreports/common/ExtentTestManager.java>
>
> You are also calling closeReport() in afterMethod which is why you get
> errors. Once you close the report, no more events can be written. See here
> <http://extentreports.relevantcodes.com/java/version2/docs.html#close-extent>
> .
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/anshooarora/extentreports/issues/187#issuecomment-173252352>
> .
>



-- 

Thanks and Regards,
Sachin Pathare
Software Engineer QA

*Vertis Microsystems LLP*
*E*:  sachin.pathare@vertisinfotech.com
*A*:  B-407 - Fourth Floor, Teerth Technospace IT Park, Off
Mumbai-Bangalore Highway, Baner, Pune, India 411045
*O*: +91 (0) 20 6720 7602 | M:+91 9011454470
*W*: www.vertisinfotech.com
LinkedIn: http://www.linkedin.com/company/vertis-microsystems-llp

*Disclaimer*: The information contained in this e-mail and any attachments
is confidential, may be privileged and is intended solely for the use of
the named addressee. Access to this e-mail by any other person is not
authorized. If you are not the intended recipient, you should not disclose,
copy, distribute, take any action or rely on it and you should, please,
notify the sender by reply.
<https://drive.google.com/a/vertisinfotech.com/folderview?id=0B1ywHyjXHOhOeHRjcllaeGdIU2s&usp=gmail>

**Comment by karthikbhatnagar** [comment]
Please use proper markdown, it is impossible to read code.  Your questions are already answered in docs.  See under test tree of src/java, you will find lots of examples created by Anshoo already

**Comment by sumahc** [comment]
Hi, 
      I am using the extent report 2.40 version. But getting the null pointer exception while generating the HTML report even though i mention the config file path properly in the code. Is there any changes required in the config file?? Please help me on this.
My code looks as below.

 String filepath=".\\Reports\\AI-MonthlyReports-"+Client_Names+".html";
 extent.loadConfig(new File("C:\\Program Files (x86)\\Selenium\\SeleniumWorkspace\\Test2\\Qlikview_Project\\extent-config.xml"));
extent = new ExtentReports(filepath, false);

**Comment by amolppatil92** [comment]
hi anshoo,
i am running a script ,script gets passed but i .html  not getting created .
thanks in advance
Please find below is the code:
package scripts;

import generic.BaseTest;

import java.lang.reflect.InvocationTargetException;
import java.util.Hashtable;
import java.util.Set;

import org.testng.ITestContext;
import org.testng.ITestResult;
import org.testng.annotations.AfterTest;
import org.testng.annotations.BeforeTest;
import org.testng.annotations.DataProvider;
import org.testng.annotations.Test;

import com.relevantcodes.extentreports.ExtentReports;
import com.relevantcodes.extentreports.ExtentTest;
import com.relevantcodes.extentreports.LogStatus;

public class CreateNewCustomer  extends BaseTest
{
  ExtentReports report;
  ExtentTest logger;
  @BeforeTest
  public void setUp() throws IllegalAccessException, IllegalArgumentException, InvocationTargetException, NoSuchMethodException, SecurityException
  {
  initializeEnviourment(); 
  }
  
  @Test(dataProvider="TestDataProvider")
  public void test(String strRun, Hashtable<String, String> dataSetValue)
  {
    report=new ExtentReports("D://UPCAutomation//Amol_WorkSpace//PracticeUPCProject//src//main//resources//Scrrenshots//FirstSamplReport.html",true);
    logger=report.startTest("Test_VerifyTitle");       
     System.out.println("STRRUN---------"+strRun);
  Set<String> keys = dataSetValue.keySet();
  for(String key: keys){
      System.out.println("Value of "+key+" is: "+dataSetValue.get(key));
  }
  }
  @AfterTest
  public void tearDown()
  {
    }
  @DataProvider(name = "TestDataProvider")
  public Object[][] getDataProvider(ITestContext context)
  {
    String dataFile = context.getCurrentXmlTest().getParameter("TestFileName");
    System.out.println("DatafileName"+dataFile);
    Object [][]testdata = loadDataProvider("Core_ST_Service_Ticket_30",dataFile);
     return testdata;
  }
}

**Comment by anshooarora** [comment]
Please open a new ticket with a correct title.  Your query is not related to the original post.

PS.  You must use `flush()` to write to the file, please see the docs for more details.

**Comment by samalpartha** [comment]
@anshooarora - Great work dude. Have an issue with have previous copies of the report, I only get the latest report after I run the suite.Is there a way I can have the past reports as well

**Comment by rohit2191** [comment]
Hello Anshoo,
   I have used extent report 2.05  in my previous now I am upgrading it to extent report 2.41. while using extent-config.xml is getting error. I added this file as after creating Extent report file path for html file and then instance for Admin_extent.loadConfig(new File("D:\\VHospital_Automation\\VHospital\\extent-config.xml")); but getting following error. Please suggest what should I do to solve this error. 
   
   symbol:   method loadConfig(File)
    [javac]   location: variable Admin_extent of type ExtentReports
    [javac] 1 error

**Comment by anshooarora** [comment]
You would have the change the "\" to "/" or "\\" and try again.

**Comment by atntusers** [comment]
Hi, I am using extentreport 2.4 version and getting below error. I added other dependencies as mentioned in above thread like xml-apis-1.4.01 but still error. please help

org.testng.TestNGException: 
Cannot instantiate class SimpleFactoryUsage.Test001
	at org.testng.internal.ObjectFactoryImpl.newInstance(ObjectFactoryImpl.java:40)
	at org.testng.internal.ClassHelper.createInstance1(ClassHelper.java:377)
	at org.testng.internal.ClassHelper.createInstance(ClassHelper.java:290)
	at org.testng.internal.ClassImpl.getDefaultInstance(ClassImpl.java:125)
	at org.testng.internal.ClassImpl.getInstances(ClassImpl.java:190)
	at org.testng.internal.TestNGClassFinder.<init>(TestNGClassFinder.java:130)
	at org.testng.TestRunner.initMethods(TestRunner.java:424)
	at org.testng.TestRunner.init(TestRunner.java:247)
	at org.testng.TestRunner.init(TestRunner.java:217)
	at org.testng.TestRunner.<init>(TestRunner.java:169)
	at org.testng.remote.support.RemoteTestNG6_9_10$1.newTestRunner(RemoteTestNG6_9_10.java:28)
	at org.testng.remote.support.RemoteTestNG6_9_10$DelegatingTestRunnerFactory.newTestRunner(RemoteTestNG6_9_10.java:61)
	at org.testng.SuiteRunner$ProxyTestRunnerFactory.newTestRunner(SuiteRunner.java:594)
	at org.testng.SuiteRunner.init(SuiteRunner.java:168)
	at org.testng.SuiteRunner.<init>(SuiteRunner.java:117)
	at org.testng.TestNG.createSuiteRunner(TestNG.java:1300)
	at org.testng.TestNG.createSuiteRunners(TestNG.java:1287)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1141)
	at org.testng.TestNG.runSuites(TestNG.java:1075)
	at org.testng.TestNG.run(TestNG.java:1047)
	at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:126)
	at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:137)
	at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:58)
Caused by: java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(Unknown Source)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(Unknown Source)
	at java.lang.reflect.Constructor.newInstance(Unknown Source)
	at org.testng.internal.ObjectFactoryImpl.newInstance(ObjectFactoryImpl.java:29)
	... 22 more
Caused by: java.lang.NoClassDefFoundError: freemarker/template/TemplateException
	at com.relevantcodes.extentreports.ExtentReports.<init>(ExtentReports.java:84)
	at com.relevantcodes.extentreports.ExtentReports.<init>(ExtentReports.java:206)
	at Factory.SimpleReportFactory.getReporter(SimpleReportFactory.java:12)
	at SimpleFactoryUsage.Test001.<init>(Test001.java:15)
	... 27 more
Caused by: java.lang.ClassNotFoundException: freemarker.template.TemplateException
	at java.net.URLClassLoader$1.run(Unknown Source)
	at java.net.URLClassLoader$1.run(Unknown Source)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.net.URLClassLoader.findClass(Unknown Source)
	at java.lang.ClassLoader.loadClass(Unknown Source)
	at sun.misc.Launcher$AppClassLoader.loadClass(Unknown Source)
	at java.lang.ClassLoader.loadClass(Unknown Source)
	... 31 more

**Comment by Ashme009** [comment]
Hi anshoo

If i'm trying to run multiple test case using testng.xml file im not able to see all the test cases in a report. its just showing me the first test case. can you please help me out in this. In this i have runn two test cases test001 and test002, its showing the report of only one. Please suggest
![capture](https://cloud.githubusercontent.com/assets/19686889/15708764/24eee362-281e-11e6-920c-504400484446.PNG)

**Comment by jaikishan1984** [comment]
Hi,

my reports are working as desired, but now i need to read the environment parameters like: UserName, OS, Java version for a remote machine, for which i will run my code locally, so i need to add the parameters name for remote machine, so how i could i do that, please suggest.

**Comment by Novatu** [comment]
Hi Anshoo,

i am getting syntax error in this below line .

extent.loadConfig(new File("E:\\Project\\jar\\Selenium\\extentreports-java-2.41.2\\extent-config.xml"));

I am trying to make my default background to dark ..so i have written some in the xml and saved in the default location. but still its not changing into the dark background.
 
So i written the above code to call the config file.

Where is the loadconfig() method is located and what is the new version of the extentreport .Can i go ahead with the extentreport 3.x 

Could you please help me in this

**Comment by Shroet** [comment]
Novatu , Please let me know if you will be manage with this issue, I will appreciate.

**Comment by vnlkumar12** [comment]
Hi Anshoo,

Could you please tell me in How many formats the extent reports can be genrated?

**Comment by GoyalKanika** [comment]
Hi Anshoo,

While I try to refer below -  
extent.AddSystemInfo("Host Name", "YourHostName")                                  .AddSystemInfo("Environment", "YourQAEnvironment")
.AddSystemInfo("Username", "YourUserName");

I get error ExtentReport.ExtentReports does not contain definition of AddSystemInfo and no extension method AddSystemInfo accepting a first arguement of type ExtentReport.ExtentReports could be found. 
I am getting the above mentioned error for extent.LogConfig as well.

I have importedbe low packages -using AventStack.ExtentReports;
using AventStack.ExtentReports.Reporter;
using System.Reflection;
using System;
using NUnit.Framework;
using NUnit.Framework.Interfaces;


I am using Visual Studio 2015.
